### PR TITLE
[HO] Handles timeouts for cvd sub command executions.

### PR DESCRIPTION
**Pull request by ser-io**

### CODE REVIEW

**Comment by jemoreira** [comment]
looks like there are build errors, please fix those

**Comment by ser-io** [comment]
the build errors were fixed

File: frontend/src/host_orchestrator/orchestrator/instancemanager.go [hunk 1]
```diff
@@ -455,24 +477,49 @@ func (e *cvdCommandExecErr) Error() string {
...
+				"############################################\n" +
+				"\n%s\n\n" +
+				"############################################\n" +
+				"## END \n" +
+				"############################################\n"
+			log.Printf(msg, b.String())
+			return &cvdCommandExecErr{c.args, b.String(), err}
+		}
+	case <-timeoutCh:
+		return &cvdCommandTimeoutErr{c.args}
```
Comment (jemoreira) [inline frontend/src/host_orchestrator/orchestrator/instancemanager.go:522]: IIUC this marks the Run as failed but doesn't cancel execution of the command. Meaning a command can actually succeed after the timeout is reached and reported?

Comment (ser-io) [reply frontend/src/host_orchestrator/orchestrator/instancemanager.go:522]: yes, I planned to cancel the whole process in a different PR as is not as straightforward as killing the process represented by `cmd`, I also need to kill all the process spawned by cvd server for this action.

**Review by jemoreira — APPROVED** [review]

**Comment by tardyp** [comment]
Hello,
Not sure how to report that, but I got some instability building the cuttlefish debs after this patch has been merged.


```
2022/11/18 08:49:24 response with error: Operation not found: operation not found: map key didn't exist
2022/11/18 08:49:24 response with error: Operation not found: operation not found: map key didn't exist
2022/11/18 08:49:24 response with error: Operation not found: operation not found: map key didn't exist
2022/11/18 08:49:24 response with error: Wait for operation timed out: waiting for operation timed out
2022/11/18 08:49:24 response with error: Wait for operation timed out: waiting for operation timed out
--- FAIL: TestWaitOperationOperationIsDone (0.00s)
    controller_test.go:137: expected <<200>>, got 503
2022/11/18 08:49:24 failed to launch cvd with error: mkdir : no such file or directory
2022/11/18 08:49:24 failed to launch cvd with error: mkdir : no such file or directory
2022/11/18 08:49:24 failed to launch cvd with error: mkdir /tmp/cuttlefishTestDir350262304/homes/cvd-1: file exists
2022/11/18 08:49:24 `cvd` execution failed with combined stdout and stderr:
############################################
## BEGIN 
############################################
--- FAIL: TestFakeBinaryMainF (0.00s)
panic: failure [recovered]
	panic: failure
goroutine 19 [running]:
testing.tRunner.func1.1(0x7bc140, 0x8a6ed0)
	/usr/lib/go-1.15/src/testing/testing.go:1072 +0x30d
testing.tRunner.func1(0xc000084600)
	/usr/lib/go-1.15/src/testing/testing.go:1075 +0x41a
panic(0x7bc140, 0x8a6ed0)
	/usr/lib/go-1.15/src/runtime/panic.go:969 +0x1b9
github.com/google/android-cuttlefish/frontend/src/host_orchestrator/orchestrator.TestFakeBinaryMainF(0xc000084600)
	/tmp/android-cuttlefish/frontend/src/host_orchestrator/orchestrator/instancemanager_test.go:626 +0x95
testing.tRunner(0xc000084600, 0x854f98)
	/usr/lib/go-1.15/src/testing/testing.go:1123 +0xef
created by testing.(*T).Run
	/usr/lib/go-1.15/src/testing/testing.go:1168 +0x2b3
############################################
## END 
############################################
2022/11/18 08:49:24 failed to launch cvd with error: exit status 2
2022/11/18 08:49:25 failed to launch cvd with error: launch cvd stage failed: cvd execution with args "start --daemon --report_anonymous_usage_stats=y --base_instance_num=1 --system_image_dir=/tmp/cuttlefishTestDir806059430/artifacts/1_foo" timed out
FAIL
FAIL	github.com/google/android-cuttlefish/frontend/src/host_orchestrator/orchestrator	1.579s
FAIL
```

a rebuild of the CI job did work although.

**Comment by ser-io** [comment]
Hello, 

Does the instability continue?  I'm not able to reproduce it locally. This failure seems related to the work I've been doing in https://github.com/google/android-cuttlefish/pull/207/commits/9026aa4437a96fb339addb567968b8c3ae37cbdf#diff-5cc36a00bde9364f6e92748509d1e8326ebb400e1de40d3c7d01881f69fb3b60 wondering whether it got caught in the middle somehow.

**Comment by tardyp** [comment]
I am not sure. this build is part of a lengthy cuttlefish pipeline which is not rebuilt that often so I can't tell if this is really flaky.
The timestamps you see in the logs are UTC and reflect whatever was in master at that time.
Github cannot find when the commit has been merged so I am not sure :-}

**Comment by ser-io** [comment]
interesting, I'm not able to reproduce the failure locally nor in the Github Actions (https://github.com/google/android-cuttlefish/actions/runs/3493305335) so will keep an eye for new updates, you can always post them here.

**Comment by jemoreira** [comment]
It seems the problem with `TestWaitOperationOperationIsDone` is that it doesn't specify a timeout, which then defaults to 0 and causes a select to randomly choose between "0 seconds has passed" and "read from channel with data".
We'll upload a fix soon.

**Pull request merged**